openscm/parameter_views.py

```diff
@@ -199,32 +149,43 @@ def get_series(self) -> Sequence[float]:
         If the parameter has child parameters (aka ``_children`` is not empty),
         the returned value will be the sum of the values of all of the child
         parameters.
+
+        Raises
+        ------
+        ParameterEmptyError
+            Parameter is empty, i.e. has not yet been written to
         """
-        if self.is_empty:
-            return self._attempt_aggregate_child_data()
+        if self._parameter._children:
+            return sum(v.get_series() for v in self._child_data_views)
+        elif self.is_empty:
+            raise ParameterEmptyError
 
         return self._timeframe_converter.convert_from(
             self._unit_converter.convert_from(self._parameter._data)
         )
 
-    def _get_child_view(self, child_parameter: _Parameter) -> ParameterView:
-        return type(self)(
-            child_parameter,
-            self._unit_converter._target,
-            self._timeframe_converter._target,
-        )
-
-    def _sum_child_data_views(self, child_data_views: list) -> Sequence[float]:
-        data = 0
-        for v in child_data_views:
-            data += v.get_series()
-
-        return data
+    def _get_child_data_views(
+        self, parameter: _Parameter
+    ) -> Sequence["TimeseriesView"]:
+        if parameter._children:
+            return sum(
+                (self._get_child_data_views(p) for p in parameter._children.values()),
+                [],
+            )
+        return [
```

Comment:
if there are no children, but we ask for children, shouldn't we throw an error rather than just return self? Or is this needed for the recursion to work?

Reply:
that's for the recursion; for the leaves of the parameter tree

Reply:
I find these methods which are named like `get_child_parameter` but then sometimes return self really confusing, is there another pattern/name we can use. Even just calling it `get_child_parameter_or_self`?

Reply:
renamed it to ``_get_data_views_for_children_or_parameter`` and made it local as we do not need it as a class method